src/wasm-lib/kcl/src/executor.rs

```diff
@@ -2821,7 +2821,7 @@ let notNull = !myNull
         assert_eq!(
             parse_execute(code2).await.unwrap_err().downcast::<KclError>().unwrap(),
             KclError::Semantic(KclErrorDetails {
-                message: "Cannot apply unary operator ! to non-boolean value: integer".to_owned(),
+                message: "Cannot apply unary operator ! to non-boolean value: number".to_owned(),
```

Comment:
Not related to your PR, but I think this would read better as "The ! operator only works with booleans, but you're trying to use it with a number." You can make this change in the current PR or I can make it later.